File: finalproject(Fontend)/lib/model/membertrip.dart
import 'package:finalproject/model/member.dart';
import 'package:finalproject/model/payment.dart';
import 'package:finalproject/model/refund.dart';
import 'package:finalproject/model/trip.dart';
class MemberTrip {
  int? memberTripId;
  DateTime? dateJoin;
  String? memberTripStatus;
  Member? participant;
  Refund? refund;
  Trip? trip;
  List<Payment>? payments;

  MemberTrip({
    this.memberTripId,
    this.dateJoin,
    this.memberTripStatus,
    this.participant,
    this.refund,
    this.trip,
    this.payments,
  });

  factory MemberTrip.fromJson(Map<String, dynamic> json) => MemberTrip(
    memberTripId: json['memberTripId'],
    dateJoin: json['dateJoin'] != null ? DateTime.parse(json['dateJoin']) : null,
    memberTripStatus: json['memberTripStatus'] ?? '',
    participant: json['participant'] != null ? Member.fromJson(json['participant']) : null,
    refund: json['refund'] != null ? Refund.fromJson(json['refund']) : null,
    trip: json['trip'] != null ? Trip.fromJson(json['trip']) : null,
    payments: json['payments'] != null
      ? List<Payment>.from(json['payments'].map((x) => Payment.fromJson(x)))
      : [],
  );

  Map<String, dynamic> toJson() => {
    'memberTripId': memberTripId,
    'dateJoin': dateJoin?.toIso8601String(),
    'memberTripStatus': memberTripStatus,
    'participant': participant?.toJson(),
    'refund': refund?.toJson(),
    'trip': trip?.toJson(),
    'payments': payments?.map((e) => e.toJson()).toList(),
  };
}

File: FinalProject(Backend)/src/main/java/com/finalproject/model/MemberTrip.java
package com.finalproject.model;

import com.fasterxml.jackson.annotation.JsonBackReference;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonManagedReference;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.ArrayList;
import java.util.Date;
import java.util.List;

@Entity
@Table(name = "membertrips")
@AllArgsConstructor
@NoArgsConstructor
@Data
@JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
public class MemberTrip {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "memberTripId")
    private Integer memberTripId;

    @Column(name = "dateJoin", nullable = false)
    private Date dateJoin;

    @Column(name = "memberTripStatus", nullable = false)
    private String memberTripStatus;

    @ManyToOne
    @JoinColumn(name = "email", nullable = false)
    @JsonManagedReference
    private Member participant;



    @ManyToOne
    @JsonBackReference
    @JoinColumn(name = "tripId", nullable = false)
    private Trip trip;

    // ใน MemberTrip.java
    @OneToMany(mappedBy = "membertrip", cascade = CascadeType.ALL, orphanRemoval = true)
    @JsonManagedReference
    private List<Payment> payments = new ArrayList<>();



}
